bot/exts/moderation/infraction/infractions.py

```diff
@@ -495,20 +495,41 @@ async def apply_ban(
             log.trace("Old tempban is being replaced by new permaban.")
             await self.pardon_infraction(ctx, "ban", user, send_msg=is_temporary)
 
+        # If user has an elevated role (staff, partner, or community), require
+        # confirmation before banning.
+        if isinstance(user, Member) and any(role.id in constants.STAFF_PARTNERS_COMMUNITY_ROLES for role in user.roles):
+            confirmation_view = _utils.StaffBanConfirmationView(
```

Comment:
Thoughts on refactoring this whole `if` block into its own function? This `apply_ban` function is growing large.